### PR TITLE
Registration tweaks and clean up

### PR DESCRIPTION
**Pull request by shaunxp20**
This PR has some tweaks and clean up to the registration flow:
- Making the example student birthday dynamic based on the the division cutoff date
- Only displaying the student birthday verbiage/example if student registration is enabled
- Making the list of disabled profile types dynamic
- Removing the `isRegistrationOpen` setting from the `StepOne` component - this check now happens in the controller, so the registration flow will never be shown if registration is closed for everyone

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -369,7 +369,7 @@
   get 'login', to: 'signins#new', as: :login
   get 'signin', to: 'signins#new', as: :signin
 
-  get "signup", to: redirect("/"), as: :signup
+  get "signup", to: "new_registration#show", as: :signup
```
Comment (shaunxp20) [inline config/routes.rb:372]: This is undoing the commit I made earlier this week when we were disabling registrations.

File: app/javascript/new_registration/components/StepOne.vue [hunk 2]
```diff
@@ -90,33 +90,51 @@ export default {
...
         label: `<img src="${require('signup/myTG-mentor.png')}" alt=""> <span class="s1-label-text">I am over 18 years old and will guide a team</span>`,
-          value: "mentor"
+          value: 'mentor'
       }
     },
     judgeProfileType() {
       return {
         label: `<img src="${require('signup/myTG-mentor.png')}" alt=""> <span class="s1-label-text">I am over 18 years old and will <span class="font-bold">judge submissions</span></span>`,
-        value: "judge"
+        value: 'judge'
```
Comment (shaunxp20) [inline app/javascript/new_registration/components/StepOne.vue:131]: Random comment - just making all the quotes in the JS in this file consistent, until we get a JS formatter in place, I just picked single quotes. 😄 😅

**Review by devkonrad — APPROVED** [review]
Greate Job, @shaunxp20! I liked a lot, the code looks very more organizated.

**Review by viviancan — APPROVED** [review]
LGTM! Did not pull down. Thanks Shaun 🎉

**Pull request merged**